Description: Add .file tags to all start.S files
 When linking raw object files, if they don't have a FILE entry _and_
 they contain local symbols, ld.bfd will add FILE entries itself, but
 with the file name specified on the command line. As the diet wrapper
 adds an absolute path for the startup file here, this causes an
 implicit encoding of the build path. Simply add .file "start.S at the
 beginning of all start.S startup files to avoid this problem.
 .
 See also: https://bugs.debian.org/857315
Author: Christian Seiler <christian@iwakd.de>
Forwarded: not-yet

--- a/aarch64/start.S
+++ b/aarch64/start.S
@@ -2,6 +2,8 @@
 #include "syscalls.h"
 #include "aarch64-features.h"
 
+.file	"start.S"
+
 /* FIXME: dynamic library support: x0 contains ldso's _fini that needs
  *        to be registered with atexit... */
 
--- a/alpha/start.S
+++ b/alpha/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .text
 .align 2
 	.set noreorder
--- a/arm/start.S
+++ b/arm/start.S
@@ -2,6 +2,8 @@
 #include "syscalls.h"
 #include "arm-features.h"
 
+.file "start.S"
+
 #ifdef __ARM_EABI__
 
 FUNC_START	_start
--- a/i386/start.S
+++ b/i386/start.S
@@ -1,6 +1,8 @@
 #include "dietfeatures.h"
 #include "PIC.h"
 
+.file "start.S"
+
 .text
 .global _start
 _start:
--- a/ia64/start.S
+++ b/ia64/start.S
@@ -13,6 +13,8 @@
 
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .psr abi64
 .psr lsb
 .lsb
--- a/mips/start.S
+++ b/mips/start.S
@@ -1,6 +1,8 @@
 #include "dietfeatures.h"
 #include "mips-asm.h"
 
+.file "start.S"
+
 #define zero 0
 #define ra 31
 #define sp 29
--- a/parisc/start.S
+++ b/parisc/start.S
@@ -13,6 +13,8 @@
 
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .level 1.1
 
 .text
--- a/ppc/start.S
+++ b/ppc/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .text
 .global _start
 .type _start,@function
--- a/ppc64/start.S
+++ b/ppc64/start.S
@@ -1,6 +1,8 @@
 #include "dietfeatures.h"
 #include "ppc64-features.h"
 
+.file "start.S"
+
 #if defined(_CALL_ELF) && _CALL_ELF == 2
 	.section ".text"
 	.globl	_start
--- a/s390/start.S
+++ b/s390/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .text
 .global _start
 _start:
--- a/s390x/start.S
+++ b/s390x/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .text
 .global _start
 _start:
--- a/sparc/start.S
+++ b/sparc/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 	.section ".text"
 	.align 4
 	.global _start
--- a/sparc64/start.S
+++ b/sparc64/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 	.section ".text"
 	.align 4
 	.global _start
--- a/x32/start.S
+++ b/x32/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .text
 .global _start
 _start:
--- a/x86_64/start.S
+++ b/x86_64/start.S
@@ -1,5 +1,7 @@
 #include "dietfeatures.h"
 
+.file "start.S"
+
 .text
 .global _start
 _start:
